packages/react-core/src/components/DualListSelector/DualListSelectorListItem.tsx

```diff
@@ -9,70 +13,78 @@ export interface DualListSelectorListItemProps extends React.HTMLProps<HTMLLIEle
   isSelected?: boolean;
-  /** Flag indicating this list item is in the chosen pane. */
-  isChosen?: boolean;
-  /** Internal callback to pass this ref up to the parent. */
-  sendRef?: (optionRef: React.ReactNode, index: number) => void;
-  /** Internal field used to keep track of the order of filtered options. */
-  filteredIndex?: number;
-  /** Internal field used to keep track of order of unfiltered options. */
-  orderIndex?: number;
   /** Callback fired when an option is selected.  */
-  onOptionSelect?: (e: React.MouseEvent | React.ChangeEvent, index: number, isChosen: boolean) => void;
-  /** ID of the option */
-  id: string;
+  onOptionSelect?: (e: React.MouseEvent | React.ChangeEvent | React.KeyboardEvent, id?: string) => void;
+  /** ID of the option. */
+  id?: string;
+  /** @hide Internal field used to keep track of order of unfiltered options. */
+  orderIndex?: number;
+  /** @hide Forwarded ref */
+  innerRef?: React.RefObject<HTMLLIElement>;
+  /** Flag indicating this item is draggable for reordring */
+  isDraggable?: boolean;
+  /** Accessible label for the draggable button on draggable list items */
+  draggableButtonAriaLabel?: string;
 }
 
-export class DualListSelectorListItem extends React.Component<DualListSelectorListItemProps> {
-  private ref = React.createRef<HTMLButtonElement>();
-  static displayName = 'DualListSelectorListItem';
-
-  componentDidMount() {
-    this.props.sendRef(this.ref.current, this.props.filteredIndex);
-  }
+export const DualListSelectorListItemBase: React.FunctionComponent<DualListSelectorListItemProps> = ({
+  onOptionSelect,
+  orderIndex,
+  children,
+  className,
+  id = getUniqueId('dual-list-selector-list-item'),
+  isSelected,
+  innerRef,
+  isDraggable = false,
+  draggableButtonAriaLabel = 'Reorder option',
+  ...props
+}: DualListSelectorListItemProps) => {
+  const ref = innerRef || React.useRef<HTMLLIElement>(null);
+  const { setFocusedOption } = React.useContext(DualListSelectorListContext);
 
-  componentDidUpdate() {
-    this.props.sendRef(this.ref.current, this.props.filteredIndex);
-  }
+  return (
+    <li
+      className={css(styles.dualListSelectorListItem, className)}
+      key={orderIndex}
+      onClick={(e: React.MouseEvent) => {
+        setFocusedOption(id);
+        onOptionSelect(e, id);
+      }}
+      onKeyDown={(e: React.KeyboardEvent) => {
+        if (e.key === ' ' || e.key === 'Enter') {
+          (document.activeElement as HTMLElement).click();
+          e.preventDefault();
+        }
+      }}
+      aria-selected={isSelected}
+      id={id}
+      ref={ref}
+      role="option"
+      tabIndex={-1}
+      {...props}
+    >
+      <div className={css(styles.dualListSelectorListItemRow, isSelected && styles.modifiers.selected)}>
+        {isDraggable && (
+          <div className={css(styles.dualListSelectorDraggable)}>
+            <Button isDisabled variant={ButtonVariant.plain} aria-label={draggableButtonAriaLabel}>
```

Comment:
why is this disabled? If the item is draggable, I would expect it to be enabled.

Reply:
I will play with it more - the issue is that we dont have any keyboard navigation implemented for drag/drop yet and the button trapping focus is pretty confusing. I had thought Zack also disabled this button in the draggable datalist - but i see now that he didn't - he only made sure no events could fire on the buttons - I'll see if I can do the same